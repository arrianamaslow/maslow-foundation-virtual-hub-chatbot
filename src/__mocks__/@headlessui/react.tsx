/**
 * @jest-environment jsdom
 */

import { DialogProps } from '@headlessui/react'
import React, { ReactNode } from 'react'

type DialogPanelTestProps = {
  children?: ReactNode
}

interface DialogTestProps extends DialogProps {
  children?: ReactNode
}

const DialogPanel: React.FC<DialogPanelTestProps> = ({ children }) => {
  return <div data-testid='mock-dialog-panel-component'>{children}</div>
}

const Dialog: React.FC<DialogTestProps> = ({ open, children }) => {
  return (
    <div data-testid='mock-dialog-component'>
      {open ? 'DIALOGUEOPENED' : 'DIALOGUECLOSED'}
      {children}
    </div>
  )
}

export { Dialog, DialogPanel }
