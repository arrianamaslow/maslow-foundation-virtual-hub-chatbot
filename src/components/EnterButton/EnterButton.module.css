.enterButton {
  @apply stroke-[#000000] hover:stroke-[#60615F];
}

.enterButton:disabled {
  @apply stroke-[#B3B6B0];
}

.enterButton:active {
  @apply stroke-[#6D6262];
}
