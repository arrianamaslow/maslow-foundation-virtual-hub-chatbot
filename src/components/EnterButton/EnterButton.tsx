import React from 'react'

import style from './EnterButton.module.css'

type EnterButtonProps = {
  isDisabled: boolean
  onClick: () => void
}

const EnterButton: React.FC<EnterButtonProps> = ({ isDisabled, onClick }) => {
  return (
    <button className={style.enterButton} disabled={isDisabled} onClick={onClick}>
      <svg
        width='32'
        height='32'
        viewBox='0 0 32 32'
        fill='none'
        xmlns='http://www.w3.org/2000/svg'
      >
        <g id='Group'>
          <path
            id='Vector'
            d='M24 1.14307H7.99997C4.21287 1.14307 1.14282 4.21311 1.14282 8.00021V24.0002C1.14282 27.7873 4.21287 30.8574 7.99997 30.8574H24C27.7871 30.8574 30.8571 27.7873 30.8571 24.0002V8.00021C30.8571 4.21311 27.7871 1.14307 24 1.14307Z'
            strokeWidth='2'
            strokeLinecap='round'
            strokeLinejoin='round'
          />
          <path
            id='Vector_2'
            d='M9.14282 16L16 8L22.8571 16'
            strokeWidth='2'
            strokeLinecap='round'
            strokeLinejoin='round'
          />
          <path
            id='Vector_3'
            d='M16 8V24'
            strokeWidth='2'
            strokeLinecap='round'
            strokeLinejoin='round'
          />
        </g>
      </svg>
    </button>
  )
}

export default EnterButton
