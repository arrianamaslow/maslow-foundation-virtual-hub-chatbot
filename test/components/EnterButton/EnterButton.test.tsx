/**
 * @jest-environment jsdom
 */
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import EnterButton from '@/components/EnterButton/EnterButton'

describe('<EnterButton />', () => {
  it('renders correctly', () => {
    const tree = render(<EnterButton isDisabled={false} onClick={() => {}} />)
    expect(tree).toMatchSnapshot()
  })
  it('fires a click event', () => {
    const clickEvent = jest.fn()
    render(<EnterButton isDisabled={false} onClick={clickEvent} />)
    const button = screen.getByRole('button')
    fireEvent.click(button)
    expect(clickEvent).toHaveBeenCalled()
  })

  it('is not clickable when disabled', () => {
    const clickEvent = jest.fn()
    render(<EnterButton isDisabled={true} onClick={clickEvent} />)
    const button = screen.getByRole('button')
    fireEvent.click(button)
    expect(clickEvent).not.toHaveBeenCalled()
  })
})
