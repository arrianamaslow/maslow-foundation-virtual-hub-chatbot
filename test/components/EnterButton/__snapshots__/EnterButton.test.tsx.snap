// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnterButton /> renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="enterButton"
      >
        <svg
          fill="none"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="Group"
          >
            <path
              d="M24 1.14307H7.99997C4.21287 1.14307 1.14282 4.21311 1.14282 8.00021V24.0002C1.14282 27.7873 4.21287 30.8574 7.99997 30.8574H24C27.7871 30.8574 30.8571 27.7873 30.8571 24.0002V8.00021C30.8571 4.21311 27.7871 1.14307 24 1.14307Z"
              id="Vector"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M9.14282 16L16 8L22.8571 16"
              id="Vector_2"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M16 8V24"
              id="Vector_3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </g>
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="enterButton"
    >
      <svg
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="Group"
        >
          <path
            d="M24 1.14307H7.99997C4.21287 1.14307 1.14282 4.21311 1.14282 8.00021V24.0002C1.14282 27.7873 4.21287 30.8574 7.99997 30.8574H24C27.7871 30.8574 30.8571 27.7873 30.8571 24.0002V8.00021C30.8571 4.21311 27.7871 1.14307 24 1.14307Z"
            id="Vector"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M9.14282 16L16 8L22.8571 16"
            id="Vector_2"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M16 8V24"
            id="Vector_3"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </g>
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
